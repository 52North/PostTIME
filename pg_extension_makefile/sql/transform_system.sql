SELECT pt_transform_system('2013-12-24','TCS001');
SELECT pt_transform_system('-4713','TCS001');
SELECT pt_transform_system('1970-1-1T0:0:0.001','TCS002');
SELECT pt_transform_system('TCS0021','CAL001');
SELECT pt_transform_system('TCS0021','TCS001');
SELECT pt_transform_system('TCS0011','TCS003');
/* error */
SELECT pt_transform_system('1970-1-1T0:0:0.001','ORD001');
SELECT pt_transform_system('ORD001Triassic','TCS002');
SELECT pt_transform_system('1970-1-1T0:0:0.001','CAL002');
SELECT pt_transform_system('1970-1-1T0:0:0.001','dasdfa');
SELECT pt_transform_system('1970-1-1T0:0:0.001','CALCAL');

