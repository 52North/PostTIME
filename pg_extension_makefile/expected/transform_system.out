SELECT pt_transform_system('2013-12-24','TCS001');
         pt_transform_system         
-------------------------------------
 TCS0012456650.500000/2456651.500000
(1 row)

SELECT pt_transform_system('-4713','TCS001');
     pt_transform_system     
-----------------------------
 TCS001-326.500000/38.500000
(1 row)

SELECT pt_transform_system('1970-1-1T0:0:0.001','TCS002');
 pt_transform_system 
---------------------
 TCS0020.001000
(1 row)

SELECT pt_transform_system('TCS0021','CAL001');
      pt_transform_system       
--------------------------------
 CAL0011970-01-01T00:00:01.000Z
(1 row)

SELECT pt_transform_system('TCS0021','TCS001');
 pt_transform_system  
----------------------
 TCS0012440587.500012
(1 row)

SELECT pt_transform_system('TCS0011','TCS003');
  pt_transform_system  
-----------------------
 TCS00386400000.000000
(1 row)

/* regular types */
SELECT pt_transform_system('R10/1970-1-1T0:1/PT1S','TCS002');
      pt_transform_system      
-------------------------------
 TCS002R10/60.000000/P1.000000
(1 row)

SELECT pt_transform_system('TCS002R10/2/P1','CAL001');
             pt_transform_system             
---------------------------------------------
 CAL001R10/1970-01-01T00:00:02.000Z/PT1.000S
(1 row)

/* error */
SELECT pt_transform_system('1970-1-1T0:0:0.001','ORD001');
ERROR:  This function does not work with Ordinal Systems
SELECT pt_transform_system('ORD001Triassic','TCS002');
    pt_transform_system     
----------------------------
 TCS002-210866759999.989990
(1 row)

SELECT pt_transform_system('1970-1-1T0:0:0.001','CAL002');
ERROR:  A calendar system with the given id is unknown.
SELECT pt_transform_system('1970-1-1T0:0:0.001','dasdfa');
ERROR:  Please specify only the new reference system.
SELECT pt_transform_system('1970-1-1T0:0:0.001','CALCAL');
ERROR:  Can not read the given reference system key.
